DragaliaAPI/Controllers/Dragalia/DungeonRecordController.cs

```diff
@@ -38,9 +38,25 @@ IUpdateDataService updateDataService
     public async Task<DragaliaResult> Record(DungeonRecordRecordRequest request)
     {
         DungeonSession session = await this.dungeonService.FinishDungeon(request.dungeon_key);
+        DbQuest? oldQuestData = await this.questRepository
+            .GetQuests(this.DeviceAccountId)
+            .SingleOrDefaultAsync(x => x.QuestId == session.DungeonId);
+
+        bool isFirstClear = oldQuestData is null || oldQuestData.PlayCount == 0;
+        bool? oldMissionClear1 = oldQuestData?.IsMissionClear1;
+        bool? oldMissionClear2 = oldQuestData?.IsMissionClear2;
+        bool? oldMissionClear3 = oldQuestData?.IsMissionClear3;
```

Comment:
I think DungeonStartController will create the quest in the db, so you could use SingleAsync and ditch the null checking

Reply:
Found out that DungeonRecordTest has a bunch of null fields passed into request so ended up keeping it to prevent them from throwing errors